backend/src/packages/auth/libs/helpers/assert-jwt-payload.helper.ts

```diff
@@ -0,0 +1,7 @@
+import { type JwtPayload } from 'shared/build';
+
+const assertJwtPayload = (payload: unknown): payload is JwtPayload => {
+  return typeof payload === 'object' && !!payload && ('id' in payload) && typeof payload.id === 'number';
```

Comment:
Can we make use of joi here?

Reply:
Yes, I think so

Reply:
However i doubt Joi is casting the value to required type as this method does.

Reply:
@mV22110702 the problem here is that this function narrows down the type to object, having id. Joi can check the value on javascript level, but on the typescript the value would still be unknown.
Although, there is a workaround for this. I can define the type of jwt globally making jwt verificator think that the type contains id by default. Then it will be convenient to check it as you suggested using Joi.

Reply:
This is not strict casting, this called narrowing, and it is in the return type:
... : payload is JwtPayload